Feature/decrypt

andrewmil opened the pull request:
Major changes in decrypt.html, views.py, test_console.py.

Potentially need to add more in the way of unit testing. The UI will be updated as more functionality is added.

Commit:
adding decrypt view+html

Commit:
adding POST request to decrypt

Commit:
/decrypt can now decrypt an encrypted string

Commit:
adding basic bootstrap to html

Commit:
adding container for css

Commit:
working on decrypt unit tests

Commit:
adding unit tests for decrypt functionality

Commit:
small changes to changelog and views

Commit:
changing button or decrypt

necrophonic commented on console/views.py:
```diff
@@ -1,8 +1,41 @@
+    if r.status_code == 404:
+        logger.error('Could not connect to ' + url, status_code=r.status_code)
+        raise Exception('404 Error')
+    else:
```
What about if `decrypt` gives you a `500` error? Should probably check the positive case (`200`) first and then deal with other errors.

Commit:
Updated exception handling

warrenbailey commented on console/views.py:
```diff
@@ -1,8 +1,57 @@
+            logger.info("Posting data to sdx-decrypt")
+            decrypt_response = send_data(url, data)
+        except ClientError:
+            return render_template('decrypt.html', decrypted_data='Client error')
```
I think it would be cleaner to have a separate variable for error messages rather than reusing decrypted data.

Also since the same template is used in each call to render_template, it might be better to have a single return statement at the end of the method.

warrenbailey commented on console/views.py:
```diff
@@ -1,8 +1,57 @@
+    except requests.exceptions.ConnectionError as e:
+        logger.error('Could not connect to sdx-decrypt', response="connection error")
+        raise e
+        return
```
return statement isn't needed here, its actually unreachable.

warrenbailey commented on console/views.py:
```diff
@@ -1,8 +1,57 @@
+        raise e
+        return
+
+    if (r.status_code > 199 and r.status_code < 300):
```
The parentheses aren't need in the if and elif statements.

It could also be simplified to:
if 199 < r.status_code < 300:

warrenbailey commented on console/views.py:
```diff
@@ -1,8 +1,57 @@
+    if request.method == "POST":
+        data = request.form['EncryptedData']
+        url = settings.SDX_DECRYPT_URL
+        decrypt_response = requests.Response()
```
You shouldn't need to construct a response object here as the call to send_data returns one. Actually I don't think this assignment will actually do anything, so you can probably just remove the line.

warrenbailey commented on console/views.py:
```diff
@@ -1,8 +1,57 @@
+    try:
+        r = requests.post(url, data)
+    except requests.exceptions.ConnectionError as e:
+        logger.error('Could not connect to sdx-decrypt', response="connection error")
```
not sure I understand the need for the response variable in the log message here?

warrenbailey commented on console/views.py:
```diff
@@ -1,8 +1,57 @@
+        logger.info('Returned from ' + url, response="ok", status_code=r.status_code)
+    elif (r.status_code > 399 and r.status_code < 500):
+        logger.error('Could not decrypt message', response="client error", status_code=r.status_code)
+        raise ClientError
```
I wonder if "r.reason" would be better than handcrafted text in the responses variable, e.g.

logger.error('Could not decrypt message', response=r.reason, status_code=r.status_code)

warrenbailey reviewed (changes requested):
I think you have some dependencies missing from the requirements.txt (i.e. flask, requests) as this doesn't build for me without manually installing those dependencies.

The change looks goods, just a few minor comments to address :)

Commit:
Various changes folowing code review

andrewmil commented:
I've think I've fixed all the small errors mentioned.

About the response variable in the log messages, I've included that because I've taken the structure from sdx-collect to try and be consistent. Maybe that's the wrong way to go though?
I liked the idea of using r.response for this variable if we are, I hadn't seen that before.

Also changed the structure for rendering the decrypt template as suggested.

Also remade requirements.txt and test_requirements.txt. Removed a lot of unnecessary stuff in there and added the new dependencies.

warrenbailey commented:
LGTM 👍
